File: src/main/java/me/ste/ivremotecontrol/block/decorinterface/decor/Decor.java
package me.ste.ivremotecontrol.block.decorinterface.decor;

import minecrafttransportsimulator.blocks.components.ABlockBase;
import minecrafttransportsimulator.jsondefs.JSONText;
import minecrafttransportsimulator.packets.components.APacketEntity;

import java.util.Collection;
import java.util.List;

public interface Decor {
    List<String> getTextLines(ABlockBase.Axis axis);
    void setTextLines(ABlockBase.Axis axis, List<String> lines);
    APacketEntity<?> getUpdatePacket(ABlockBase.Axis axis, List<String> lines);
    List<JSONText> getDefinitionTextObjects(ABlockBase.Axis axis);
    boolean isPole();
}

File: src/main/kotlin/me/ste/ivremotecontrol/block/decorinterface/decor/Decor.kt
package me.ste.ivremotecontrol.block.decorinterface.decor

import minecrafttransportsimulator.blocks.components.ABlockBase
import minecrafttransportsimulator.jsondefs.JSONText
import minecrafttransportsimulator.packets.components.APacketEntity

abstract class Decor {
    abstract fun getTextLines(axis: ABlockBase.Axis): Map<JSONText, String>?
    abstract fun setTextLines(axis: ABlockBase.Axis, lines: List<String>)
    abstract fun getUpdatePacket(axis: ABlockBase.Axis, lines: List<String>): APacketEntity<*>?
    abstract val isPole: Boolean
}